py/picca/delta_extraction/expected_fluxes/true_continuum.py

```diff
@@ -307,76 +307,67 @@ def read_true_continuum(self, forest):
-            mean_optical_depth = np.ones(forest.log_lambda.size)
-            tau, gamma, lambda_rest_frame = 0.0023, 3.64, 1215.67
-            w = 10.**forest.log_lambda / (1. + forest.z) <= lambda_rest_frame
-            z = 10.**forest.log_lambda / lambda_rest_frame - 1.
+            forest.continuum *= self.get_mean_flux(10**forest.log_lambda)[0]
 
         elif Forest.wave_solution == "lin":
             forest.continuum = true_continuum(forest.lambda_)[0]
-            mean_optical_depth = np.ones(forest.lambda_.size)
-            tau, gamma, lambda_rest_frame = 0.0023, 3.64, 1215.67
-            w = forest.lambda_ / (1. + forest.z) <= lambda_rest_frame
-            z = forest.lambda_ / lambda_rest_frame - 1.
+            forest.continuum *= self.get_mean_flux(forest.lambda_)
         else:
             raise ExpectedFluxError("Forest.wave_solution must be either 'log' "
                                     "or 'lin'")
 
-        mean_optical_depth[w] *= np.exp(-tau * (1. + z[w])**gamma)
-        forest.continuum *= mean_optical_depth
-        forest.hpcont = healpix
         return forest
 
-    def read_var_lss(self):
-        """Read the LSS delta variance from files (written by the raw analysis)
+    def read_raw_statistics(self):
+        """Read the LSS delta variance and mean transmitted flux from files written by the raw analysis
         """
-        #var_lss files are only for lya so far, this will need to be updated so that regions other than Lya are available
-
-        if self.var_lss_binning == 'log':
-            filename = 'colore_v9_lya_log.fits.gz'
-        elif self.var_lss_binning == 'lin_2.4':
-            filename = 'colore_v9_lya_lin_2.4.fits.gz'
-        elif self.var_lss_binning == 'lin_3.2':
-            filename = 'colore_v9_lya_lin_3.2.fits.gz'
-        else:
-            self.logger.info(f'Trying to use costume var_lss file from {self.var_lss_binning}')
-            filename = self.var_lss_binning
+        #files are only for lya so far, this will need to be updated so that regions other than Lya are available
 
-        if filename == self.var_lss_binning:
-            var_lss_file = self.var_lss_binning
+        if self.raw_statistics_filename is not None:
+            filename = self.raw_statistics_filename
         else:
-            var_lss_file = resource_filename('picca', 'delta_extraction')
-            var_lss_file += '/expected_fluxes/var_lss/' + filename
+            filename = resource_filename('picca', 'delta_extraction') + '/expected_fluxes/var_lss/'
+            if self.raw_statistics_binning == 'log':
+                filename += 'colore_v9_lya_log.fits.gz'
+            elif self.raw_statistics_binning == 'lin_2.4':
+                filename += 'colore_v9_lya_lin_2.4.fits.gz'
+            elif self.raw_statistics_binning == 'lin_3.2':
+                filename += 'colore_v9_lya_lin_3.2.fits.gz'
+        self.logger.info(f'Reading raw statistics var_lss and mean_flux from file: {filename}')
```

Comment:
I guess here you are trying to guess which file to use, if the user has not provided the raw_stats file.

However, why do you ask for an extra variable (raw_stats_binning) instead of checking which binning you are actually using in the true continuum analysis? These should always be synchronised anyway, so no need to specify two different pixelisations.

Reply:
This is because the pixelization in the true continuum analysis could be different from the ones available in raw_statistics_files. There are two options: 
- providing a raw_statistics_filename in the configuration: will directly read this file
- providing a pixelization scheme: will read the stored raw_statistics_files with the correspondent pixelization.

The second option is to allow for computing true deltas without having to compute a compatible raw analysis.

Reply:
In my opinion, if the pixelisation in the true continuum analysis is not the same that was used in the raw analysis, the code should crash.

Reply:
Yes, having these two options is clearer. It's ok, let's leave it as it is for now, as long as there is a check for consistency between the pixels (and restframe wavelength) used, I'm ok with it.

Reply:
Yes @cramirezpe that was always the idea to have to two options, at the moment I just left it as one single argument so I didn't break anything as we wanted to merge. thanks for splitting. I agree we need a check for consistency in any case.

Reply:
Ok, let's merge as is and let's open an issue to improve this later on. Thanks both!